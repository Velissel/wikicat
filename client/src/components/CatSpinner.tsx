import React from 'react';
// @ts-ignore
import styles from './CatSpinner.module.scss';
// @ts-ignore
import logo from '../logo.svg';

interface CatSpinnerProps {
  label?: string;
}
export default function CatSpinner(props: CatSpinnerProps) {
  return (
    <div className={styles.spinner}>
      <div className={styles['spinner-header']}>
        <img src={logo} className={styles['spinner-logo']} alt="logo" />
        <p>{!props.label ? 'Loading...' : props.label}</p>
      </div>
    </div>
  );
}
