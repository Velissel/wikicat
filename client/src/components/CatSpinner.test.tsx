import React from 'react';
import { render, screen } from '@testing-library/react';
import CatSpinner from './CatSpinner';

describe('CatSpinner component', () => {
  it('should render loading text by default', () => {
    const { container } = render(<CatSpinner />);
    expect(screen.getByText('Loading...')).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });

  it('should render custom text if label is supplied', () => {
    const { container } = render(<CatSpinner label="Custom" />);
    expect(screen.getByText('Custom')).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });
});
