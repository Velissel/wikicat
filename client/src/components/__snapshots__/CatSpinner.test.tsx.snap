// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CatSpinner component should render custom text if label is supplied 1`] = `
<div>
  <div
    class="spinner"
  >
    <div
      class="spinner-header"
    >
      <img
        alt="logo"
        class="spinner-logo"
        src="logo.svg"
      />
      <p>
        Custom
      </p>
    </div>
  </div>
</div>
`;

exports[`CatSpinner component should render loading text by default 1`] = `
<div>
  <div
    class="spinner"
  >
    <div
      class="spinner-header"
    >
      <img
        alt="logo"
        class="spinner-logo"
        src="logo.svg"
      />
      <p>
        Loading...
      </p>
    </div>
  </div>
</div>
`;
